legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/auth/GitLabPersonalAccessTokenSession.java

```diff
@@ -0,0 +1,154 @@
+                this.getCreationTime().equals(that.getCreationTime()) &&
+                this.tokenManager.equals(that.tokenManager);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return getCreationTime().hashCode() ^ getUserId().hashCode() ^ this.tokenManager.hashCode();
+    }
+
+    @Override
+    public Set<GitLabMode> getValidModes()
+    {
+        return this.tokenManager.getValidModes();
+    }
+
+    @Override
+    public boolean isValidMode(GitLabMode mode)
+    {
+        return this.tokenManager.isValidMode(mode);
+    }
+
+    @Override
+    public boolean gitLabOAuthCallback(GitLabMode mode, String code)
+    {
+        return this.tokenManager.gitLabOAuthCallback(mode, code);
+    }
+
+    @Override
+    public GitLabToken getGitLabToken(GitLabMode mode)
+    {
+        return this.tokenManager.getGitLabToken(mode);
+    }
+
+    @Override
+    public void clearGitLabTokens()
+    {
+        this.tokenManager.clearGitLabTokens();
+    }
+
+    @Override
+    public void putGitLabToken(GitLabMode mode, String token)
+    {
+        this.tokenManager.putPrivateAccessToken(mode, token);
+    }
+
+    @Override
+    public void putGitLabToken(GitLabMode mode, GitLabToken token)
+    {
+        if (token.getTokenType().equals(TokenType.PRIVATE))
+        {
+            this.tokenManager.putGitLabToken(mode, token);
+        }
+    }
+
+    @Override
+    public GitLabModeInfo getModeInfo(GitLabMode mode)
+    {
+        return this.tokenManager.getModeInfo(mode);
+    }
+
+    @Override
+    public Token.TokenBuilder encode(Token.TokenBuilder builder)
+    {
+        return this.tokenManager.encode(super.encode(builder));
+    }
+
+    @Override
+    protected void writeToStringInfo(StringBuilder builder)
+    {
+        this.tokenManager.appendTokenInfo(builder.append(' '));
+    }
+
+    private static GitLabTokenManager possiblyInitializeTokenManager(GitLabTokenManager tokenManager, GitlabPersonalAccessTokenProfile profile)
+    {
+        if ((tokenManager != null) && (profile != null))
+        {
+            LOGGER.debug("initializing with GitlabPersonalAccessTokenProfile: {}", profile);
+            String token = profile.getPersonalAccessToken();
+
+            // Private token is attached to PROD mode only
+            if (token != null && tokenManager.isValidMode(GitLabMode.PROD))
+            {
+                tokenManager.getValidModes()
+                        .stream()
+                        .filter(mode -> mode.equals(GitLabMode.PROD) && tokenManager.getGitLabToken(mode) == null)
+                        .map(tokenManager::getModeInfo)
+                        .forEach(modeInfo ->
+                        {
+                            tokenManager.putOAuthToken(modeInfo.getMode(), token);
+                            LOGGER.debug("Storing private access token from profile for mode {}", modeInfo.getMode());
+                        });
+            }
```

Comment:
We need to make use of the GitLab host the token was validated against here. If the validation host is the same as the host for one of the modes, then we can store the token for that mode. If there's no matching mode, then we can't store the token.